### PR TITLE
Gradle compatibility test

### PR DESCRIPTION
**Pull request by mnonnenmacher**
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/368)

### CODE REVIEW

File: analyzer/src/funTest/kotlin/GradleTest.kt [hunk 1]
```diff
@@ -88,5 +107,48 @@ class GradleTest : StringSpec() {
...
+                result shouldNotBe null
+                result!!.errors should beEmpty()
+                yamlMapper.writeValueAsString(result) shouldBe expectedResult
+            }
+        }.config(tags = setOf(ExpensiveTag))
+    }
+
+    private fun gradleWrapper(version: String) {
+        println("Installing Gradle wrapper version $version.")
+        val gradleWrapper = if (OS.isWindows) Gradle.wrapper else "./${Gradle.wrapper}"
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/kotlin/GradleTest.kt]: @sschuberth : Do you have an idea why this does not work on Windows?

Comment (sschuberth) [reply analyzer/src/funTest/kotlin/GradleTest.kt]: Yes, this is something interesting I realized lately when bootstrapping scanners: Windows requires the *absolute* path to the batch file here. This works for me:

    private fun gradleWrapper(version: String) {
        println("Installing Gradle wrapper version $version.")

        // When calling Windows batch files directly (without passing them to "cmd" as an argument), Windows requires
        // the absolute path to the batch file to be passed to the underlying ProcessBuilder for some reason.
        val wrapperAbsolutePath = File(projectDir, Gradle.wrapper).absolutePath

        ProcessCapture(projectDir, wrapperAbsolutePath, "wrapper", "--gradle-version=$version",
                "--distribution-type=ALL").requireSuccess()
    }

Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/GradleTest.kt]: Thanks, I have updated the PR.

**Comment by sschuberth** [comment]
@mnonnenmacher, now `import com.here.ort.utils.OS` is unused.

File: analyzer/src/funTest/kotlin/GradleTest.kt (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/GradleTest.kt]: Still failing, I'm trying now if `projectDir.absoluteFile` helps.

**Comment by mnonnenmacher** [comment]
@sschuberth : Thanks, fixed.

**Comment by sschuberth** [comment]
> Still failing, I'm trying now if `projectDir.absoluteFile` helps.

I doubt that, as the path is already made absolute in the end. Also, it looks like calling `gradlew.bat` actually worked. The `IOException` is not because Gradle exited with a code != 0. See how we had before

    com.here.ort.analyzer.GradleTest > initializationError FAILED
        java.io.IOException: Cannot run program "gradlew.bat"

and now we have

    com.here.ort.analyzer.GradleTest > initializationError FAILED
        java.io.IOException: Running 'C:\projects\oss-review-toolkit\analyzer\src\funTest\assets\projects\synthetic\gradle\gradlew.bat wrapper --gradle-version=4.5.1 --distribution-type=ALL' in directory 'src\funTest\assets\projects\synthetic\gradle' failed with exit code 1:

Still, I cannot reproduce the issue locally. It work for me with my original code, both from the IDE and from the command line.

**Comment by sschuberth** [comment]
@mnonnenmacher, should I take over the branch as I'm probably in a better position to debug the Windows issue?

**Comment by mnonnenmacher** [comment]
@sschuberth : Yes, please :)

**Comment by sschuberth** [comment]
@mnonnenmacher, fixed. The final issue was with Java 9 (which we are using on Windows CI), not with Windows itself.

**Comment by mnonnenmacher** [comment]
Thanks for finding the solution, you have to approve the PR, I cannot because I created it.

**Pull request merged**